napari/utils/events/types.py

```diff
@@ -1,8 +1,18 @@
-from typing import Protocol, runtime_checkable
+from typing import Any, Protocol, runtime_checkable
 
 from .event import EmitterGroup
 
 
 @runtime_checkable
 class SupportsEvents(Protocol):
+    # note that if this gets *any* object with an `events` attribute, it will pass
+    # always because type hints are not checked!
+    # this happened with some dask objects, so this is a weak spot in some cases
     events: EmitterGroup
```

Comment:
```suggestion
class SupportsEvents:
    @classmethod
    def __instancecheck__(cls, instance):
        return hasattr(instance, "events") and isinstance(instance.events, EmitterGroup)
    
    events: EmitterGroup

```

Reply:
Awesome! Butdon't we lose the benefit of static type checkers and `Protocols`?

Reply:
I do not test it with static type checkers. Maybe it could be mixed with protocols.